### DISCUSSION TITLE
why does poppler install tmux and vim

### DISCUSSION THREAD

**Discussion by boompig** [Homebrew/discussions#2073] [Everyday usage]
It also installs `gdb` (a debugger). In theory it's just a PDF editing tool. Why is it installing tools for creating interactive remote sessions?

**Comment by SMillerDev** [comment]
You can check using `brew deps --tree poppler`. It doesn't require tmux or vim so won't install those, but it does require `gdbm` as it's a python dependency, which is pulled in by glib and qt, which poppler does depend on.

**Comment by gromgit** [comment]
I suspect you already had `tmux`, `vim` and `gdb` installed, _and_ you haven't run `brew upgrade` in a while, so what you're seeing is an _upgrade_ of those formulae.

`brew install XYZ` will install dependencies that aren't already installed, and _upgrade dependencies that are already installed_. That second part will trigger a recursive upgrade of all installed formulae that depend on these just-upgraded formulae. https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-also-upgrade-a-bunch-of-other-stuff
